wandb/internal/sender.py

```diff
@@ -186,7 +197,7 @@ def handle_final(self, data):
         if data.control.req_resp:
             logger.info("informing user process exit was handled")
             # TODO: send something more than an empty result
-            resp = wandb_internal_pb2.ResultRecord()
+            resp = wandb_internal_pb2.Result()
```

Comment:
Not introduced in your change, but I'm noticing here there's a chance you put two responses in the queue (see line 190). Is that intentional?

Reply:
thanks, i removed this, it was unexecuted code because req_resp was not set.
I have plans to implement a state machine to allow for more general defer logic to allow for multi-stage handlers.